dashboard/app/models/script.rb

```diff
@@ -177,6 +177,33 @@ def self.artist_script
     Script.get_from_cache(Script::ARTIST_NAME)
   end
 
+  # Get the set of scripts that are valid for the current user, ignoring those
+  # that are hidden based on the user's permission.
+  # @param [User] user
+  # @return [AssignableInfo[]]
+  def self.valid_scripts(user)
```

Comment:
This looks right to me. However, I'm not sure this is as necessary now that we're in dashboard.

Dashboard already has a script cache. Theoretically, we could access that script cache and generate all of our assignable_info without needing to hit the db again (I think). Not sure if we want to play around with this though - it might make sense to just stick with the current approach.